principles.md

```diff
@@ -0,0 +1,25 @@
+## Product and Engineering Principles
+
+**Simpler always wins in the end**
```

Comment:
In my mind this approach has led to more problems than it has solved, even if it was just miss-applied to justify engineering design weaknesses.

Reply:
May be a wrong tagline, but do you disagree with the elaboration below?

Reply:
I only agree with the elaboration as it applies to security primitives and threat modelling. For everything else in my view there should be necessary complexity so that the product is understandable and relatable, and what we call simple tends to actually imply hidden complexities which is why I'm generally against the approach of simple as a engineering principle.

Reply:
@knisbet great points, will try to reword it to match closer to reality. I will try something like "Excess complexity stays in the way of security and adoption".

Reply:
Related I think: there is probably a good engineering corollary to Newton's first rule of scientific reasoning:

> We are to admit no more causes of natural things than such as are both true and sufficient to explain their appearances. To this purpose the philosophers say that Nature does nothing in vain, and more is in vain when less will serve; for Nature is pleased with simplicity, and affects not the pomp of superfluous causes.

science: no more complex but no simpler than the simplest system that explains observed phenomena

engineering: no more complex but no simpler than the simplest system that achieves desired phenomena